extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1116,22 +1152,20 @@ private void killTask(final String id)
 
   private DateTime getTaskStartTime(final String id)
   {
-    DateTime startTime = null;
-    TaskLocation taskLocation = getTaskLocation(id);
-    if (taskLocation != null) {
-      startTime = taskClient.getStartTime(taskLocation, id);
+    if (taskInfoProvider.getTaskLocation(id) != null) {
```

Comment:
Yes, but that's fine, when it calls into KafkaIndexTaskClient it'll make the call again right before constructing the request and if it fails at that time it'll throw the NoTaskLocationException. This is to minimize calls into the client and minimize the exceptions thrown.